Segmentation violation on gh cs create

### Describe the bug

`gh cs create` paniced on a 404 Not Found response

### Steps to reproduce the behavior

1. Type this '...'
2. View the output '....'
3. See error

### Expected vs actual behavior

A clear and concise description of what you expected to happen and what actually happened.

### Logs

```
% gh cs create -r OWNER/NAME
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x18fe28e]

goroutine 1 [running]:
github.com/cli/cli/v2/internal/codespaces/api.(*API).CreateCodespace(0x0?, {0x2af5ba0, 0xc00019c008}, 0xc00019c008?)
	github.com/cli/cli/v2/internal/codespaces/api/api.go:708 +0x1ce
github.com/cli/cli/v2/pkg/cmd/codespace.(*App).Create(0xc0001c89c0, {0x2af5ba0?, 0xc00019c008}, {{0x7ff7bfeffb90, 0xd}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...})
	github.com/cli/cli/v2/pkg/cmd/codespace/create.go:250 +0xe53
github.com/cli/cli/v2/pkg/cmd/codespace.newCreateCmd.func1(0xc0001ca780?, {0xc000796100?, 0x0?, 0x2?})
	github.com/cli/cli/v2/pkg/cmd/codespace/create.go:80 +0x65
github.com/spf13/cobra.(*Command).execute(0xc0001ca780, {0xc0007960e0, 0x2, 0x2})
	github.com/spf13/cobra@v1.4.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0xc00017c000)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
main.mainRun()
	github.com/cli/cli/v2/cmd/gh/main.go:222 +0xc7a
main.main()
	github.com/cli/cli/v2/cmd/gh/main.go:50 +0x19
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/codespaces#update-a-codespace-for-the-authenticated-user"
}
gh: Not Found (HTTP 404)
```

```
% gh --version
gh version 2.13.0 (2022-06-22)
https://github.com/cli/cli/releases/tag/v2.13.0
```

--- a/internal/codespaces/api/api.go
+++ b/internal/codespaces/api/api.go
@@ -706,7 +706,7 @@ type CreateCodespaceParams struct {
 // fails to create.
 func (a *API) CreateCodespace(ctx context.Context, params *CreateCodespaceParams) (*Codespace, error) {
 	codespace, err := a.startCreate(ctx, params)
-	if err != errProvisioningInProgress {
+	if !errors.Is(err, errProvisioningInProgress) {
 		return codespace, err
 	}
 
@@ -802,7 +802,17 @@ func (a *API) startCreate(ctx context.Context, params *CreateCodespaceParams) (*
 	defer resp.Body.Close()
 
 	if resp.StatusCode == http.StatusAccepted {
-		return nil, errProvisioningInProgress // RPC finished before result of creation known
+		b, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, fmt.Errorf("error reading response body: %w", err)
+		}
+
+		var response Codespace
+		if err := json.Unmarshal(b, &response); err != nil {
+			return nil, fmt.Errorf("error unmarshaling response: %w", err)
+		}
+
+		return &response, errProvisioningInProgress // RPC finished before result of creation known
 	} else if resp.StatusCode == http.StatusUnauthorized {
 		var (
 			ue       AcceptPermissionsRequiredError

--- a/internal/codespaces/api/api_test.go
+++ b/internal/codespaces/api/api_test.go
@@ -66,47 +66,93 @@ func createFakeListEndpointServer(t *testing.T, initalTotal int, finalTotal int)
 	}))
 }
 
-func createFakeCreateEndpointServer(t *testing.T, initalTotal int, finalTotal int) *httptest.Server {
+func createFakeCreateEndpointServer(t *testing.T, wantStatus int) *httptest.Server {
+	t.Helper()
 	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.URL.Path != "/user/codespaces" {
-			t.Fatal("Incorrect path")
+		// create endpoint
+		if r.URL.Path == "/user/codespaces" {
+			body := r.Body
+			if body == nil {
+				t.Fatal("No body")
+			}
+			defer body.Close()
+
+			var params startCreateRequest
+			err := json.NewDecoder(body).Decode(&params)
+			if err != nil {
+				t.Fatal("error:", err)
+			}
+
+			if params.RepositoryID != 1 {
+				t.Fatal("Expected RepositoryID to be 1. Got: ", params.RepositoryID)
+			}
+
+			if params.IdleTimeoutMinutes != 10 {
+				t.Fatal("Expected IdleTimeoutMinutes to be 10. Got: ", params.IdleTimeoutMinutes)
+			}
+
+			if *params.RetentionPeriodMinutes != 0 {
+				t.Fatal("Expected RetentionPeriodMinutes to be 0. Got: ", *params.RetentionPeriodMinutes)
+			}
+
+			response := Codespace{
+				Name: "codespace-1",
+			}
+
+			if wantStatus == 0 {
+				wantStatus = http.StatusCreated
+			}
+
+			data, _ := json.Marshal(response)
+			w.WriteHeader(wantStatus)
+			fmt.Fprint(w, string(data))
+			return
 		}
 
-		body := r.Body
-		if body == nil {
-			t.Fatal("No body")
-		}
-		defer body.Close()
-
-		var params startCreateRequest
-		err := json.NewDecoder(body).Decode(&params)
-		if err != nil {
-			t.Fatal("error:", err)
-		}
-
-		if params.RepositoryID != 1 {
-			t.Fatal("Expected RepositoryID to be 1. Got: ", params.RepositoryID)
+		// get endpoint hit for testing pending status
+		if r.URL.Path == "/user/codespaces/codespace-1" {
+			response := Codespace{
+				Name:  "codespace-1",
+				State: CodespaceStateAvailable,
+			}
+			data, _ := json.Marshal(response)
+			w.WriteHeader(http.StatusOK)
+			fmt.Fprint(w, string(data))
+			return
 		}
 
-		if params.IdleTimeoutMinutes != 10 {
-			t.Fatal("Expected IdleTimeoutMinutes to be 10. Got: ", params.IdleTimeoutMinutes)
-		}
-
-		if *params.RetentionPeriodMinutes != 0 {
-			t.Fatal("Expected RetentionPeriodMinutes to be 0. Got: ", *params.RetentionPeriodMinutes)
-		}
-
-		response := Codespace{
-			Name: "codespace-1",
-		}
-
-		data, _ := json.Marshal(response)
-		fmt.Fprint(w, string(data))
+		t.Fatal("Incorrect path")
 	}))
 }
 
 func TestCreateCodespaces(t *testing.T) {
-	svr := createFakeCreateEndpointServer(t, 200, 200)
+	svr := createFakeCreateEndpointServer(t, http.StatusCreated)
+	defer svr.Close()
+
+	api := API{
+		githubAPI: svr.URL,
+		client:    &http.Client{},
+	}
+
+	ctx := context.TODO()
+	retentionPeriod := 0
+	params := &CreateCodespaceParams{
+		RepositoryID:           1,
+		IdleTimeoutMinutes:     10,
+		RetentionPeriodMinutes: &retentionPeriod,
+	}
+	codespace, err := api.CreateCodespace(ctx, params)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if codespace.Name != "codespace-1" {
+		t.Fatalf("expected codespace-1, got %s", codespace.Name)
+	}
+}
+
+func TestCreateCodespaces_Pending(t *testing.T) {
+	svr := createFakeCreateEndpointServer(t, http.StatusAccepted)
 	defer svr.Close()
 
 	api := API{

